### PR TITLE
Add missing structure scales for growth of Symbiotic and Tree hulls

### PR DESCRIPTION
**Pull request by Oberlus**
Fixes missing structure growth scale for Symbiotic Hull and Tree Body hull.

@agrrr3 please, review

This should be cherry-picked for release since it solves a problem with PR #2908 which was merged into release branch.

### CODE REVIEW

**Comment by geoffthemedio** [comment]
SH_SYMBIOTIC makes sense, but why SH_TREE_BODY? There are other monster body hulls as well...

**Comment by Oberlus** [comment]
> SH_SYMBIOTIC makes sense, but why SH_TREE_BODY? There are other monster body hulls as well...

Structure of hulls and parts have not been changed in the FOCS files but in ShipHull.cpp and ShipPart.cpp through RULE_SHIP_STRUCTURE_FACTOR (e.g. SH_BASIC_MEDIUM has still `structure = 10` and AR_STD_PLATE has `capacity = 6`, but with scaling factor 8 the starting Frigate has 128 structure).
However, the calls to SetMaxStructure and SetStructure within FOCS files need to explicitly use `[[SHIP_STRUCTURE_FACTOR]]`, otherwise their growth or repair rates does not correspond to the re-scaled values.

In the case of the Dyson Forest, the +3 structure gain per turn from age should correspond to +24 with current scale factor of 8, otherwise it becomes much weaker.


I've spotted one more issue: repair rate of orbital drydocks has not been scaled and so repairing a frig takes several turns instead of usual one. Should I fix it in this PR or make a new one?

**Comment by agrrr3** [comment]
Code looks good. Monsters did not get enough love, it's quite possible that I missed to adjust some. But I am surprised about the symbiotic hull, I thought I adjusted all the organic growth.
I would totally suggest a balance pass for 0.4.10

Afaicr orbital drydock should be already ok as it depends on percent of current meter and max meter. If it doesn't work I need to have a look

**Comment by Oberlus** [comment]
> Afaicr orbital drydock should be already ok as it depends on percent of current meter and max meter. If it doesn't work I need to have a look

I think this suffices:
```
// Minimum repair per turn, at target happiness
CONST_ORB_DRYDOCK_MIN_REPAIR
'''30.0 * [[SHIP_STRUCTURE_FACTOR]]'''
```
Also some tweak in Logistics Facilitator. Already done.

Is it OK if I remove the unneeded includes of /scripting/common/misc.macros in the organic hulls since they all get it from including organic.macros?

**Comment by geoffthemedio** [comment]
> Is it OK if I remove the unneeded includes

should be fine

**Comment by agrrr3** [comment]
> > Afaicr orbital drydock should be already ok as it depends on percent of current meter and max meter. If it doesn't work I need to have a look
> 
> I think this suffices:
> 
> ```
> // Minimum repair per turn, at target happiness
> CONST_ORB_DRYDOCK_MIN_REPAIR
> '''30.0 * [[SHIP_STRUCTURE_FACTOR]]'''
> ```

Yes, looks good

**Comment by agrrr3** [comment]
> Also some tweak in Logistics Facilitator. Already done.
> 

Not pushed yet I guess

**Comment by Oberlus** [comment]
> > Also some tweak in Logistics Facilitator. Already done.
> 
> Not pushed yet I guess

Pushed, ready for review.

I also removed the old "STRUCTURE_MULTIPLICATOR" from ship_hull.macros, not used in any file within default folder.

File: default/scripting/ship_hulls/robotic/SH_LOGITICS_FACILITATOR.focs.txt [hunk 1]
```diff
@@ -81,7 +81,7 @@ Hull
             ]
             stackinggroup = "FLAGSHIP_EFFECT_LOGISTICS"
             priority = [[VERY_LATE_PRIORITY]]
-            effects = SetStructure value = max((Target.Structure + ( Target.Structure/10 )), (Target.Structure + 20))
+            effects = SetStructure value = max((Target.Structure + ( Target.Structure/10 )), (Target.Structure + 20*[[SHIP_STRUCTURE_FACTOR]]))
```
Comment (agrrr3) [inline default/scripting/ship_hulls/robotic/SH_LOGITICS_FACILITATOR.focs.txt]: add parenthesis around 20*[[...]]

Comment (agrrr3) [reply default/scripting/ship_hulls/robotic/SH_LOGITICS_FACILITATOR.focs.txt]: also the scaling probably should be done the same way for the other +50 effect above. Also it seems a bit fishy that repairs get a boost if there is combat in the system.

Comment (Oberlus) [reply default/scripting/ship_hulls/robotic/SH_LOGITICS_FACILITATOR.focs.txt]: Fixed

**Comment by geoffthemedio** [comment]
@agrrr3 OK now?

**Comment by agrrr3** [comment]
Code Looks good to merge

**Comment by Vezzra** [comment]
Needs rebase.

**Comment by Oberlus** [comment]
> Needs rebase.

I have doubts on how to do this.
The conflicts are in the file default/scripting/fields/FLD_NANITE_SWARM.focs.txt
Easy to solve, but this PR is also intended for cherry-pick into release-v0.4.10, which doesn't have that file. And, following o01eg's advice from a previous PR (as I understood it), I make PRs with a single commit for all files changed.

What should I do? Fix conflict for merge into master and make another PR for v0.4.10?

**Comment by Oberlus** [comment]
Oh, sorry... I tried web based fix conflict and it did a merge...
How can I undo this to make the proper rebase from my local machine? (my local branch is untouched)

Edit: merge-instead-of-rebase problem solved.
I still need advice about what to do for the cherry-pick into release issue.

**Comment by adrianbroher** [comment]
> Oh, sorry... I tried web based fix conflict and it did a merge...

You can force push your local branch to the PR.

**Comment by Vezzra** [comment]
> I still need advice about what to do for the cherry-pick into release issue.

In such cases tag the PR as cherry-pick for release as usual, and do one of these:

- Either leave a comment describing the expected merge conflicts with the release branch, and suggestions on how to resolve them (in this case "don't merge `default/scripting/fields/FLD_NANITE_SWARM.focs.txt`"). I'll take it from there.

- Or, if you feel up to it, do the cherry picking yourself and resolve the merge conflicts. Less work for me, and therefore much appreciated, *but only if you're really sure what you're doing*!!! Otherwise it's most likely a lot more work for me, and therefore more appreciated if you leave the actual cherry picking and merge conflict resolving to me.

Take your pick and let me know. 😃

**Comment by Oberlus** [comment]
@Vezzra 

I guess I didn't know what I was doing:

I did the cherrypicking on my local release-v0.4.10 branch, resolved the conflict and pushed it to my fork. Now it is a new PR #3060 intended to merge into release-v0.4.10.
I don't know if this is what you were expecting from the second option you propose.

If it isn't, then the first option you proposed is the way to go (to be more specific, `git rm default/scripting/fields/FLD_NANITE_SWARM.focs.txt` when cherripicking 0eaea4e).
I can't tag PRs, AFAIK, so someone else should do it.

**Comment by Vezzra** [comment]
> I did the cherrypicking on my local release-v0.4.10 branch, resolved the conflict and pushed it to my fork. Now it is a new PR #3060 intended to merge into release-v0.4.10.
I don't know if this is what you were expecting from the second option you propose.

Not exactly, although certainly also a way how to do it, just overkill in this particular case IMO. 

Providing a separate PR for the release branch would be the way to go if the differences between what goes into master and what goes into the release branch (so, basically, the differences you need to introduce when resolving the merge conflicts) are so big/complicated that they should be reviewed seperately before merging them into the release branch.

So, in this case, as you said, the first option is the way to go, as you apparently do not have commit rights, which are required for the second option. Sorry, I totally forgot to take that into account.

Meaning, once this gets merged, whoever does the merge please just mark the PR for cherry picking.

#3060 can be closed.

**Pull request merged**